src/lean_spec/subspecs/containers/state/state.py

```diff
@@ -439,41 +446,47 @@ def process_attestations(
                 continue  # Skip invalid votes
 
             # Check if source checkpoint is justified.
-            source_slot_int = source.slot.as_int()
-            target_slot_int = target.slot.as_int()
+            source_slot = source.slot.as_int()
+            target_slot = target.slot.as_int()
 
             # Ensure we have enough justified slots history.
-            if source_slot_int < len(justified_slots):
-                source_is_justified = justified_slots[source_slot_int]
-            else:
-                continue  # Source is too far in the past
+            if source_slot >= len(justified_slots):
+                continue
 
+            source_is_justified = justified_slots[source_slot]
+
+            target_tracked = target_slot < len(justified_slots)
```

Comment:
@GrapeBaBa  lets not make any other changes in this PR so as to keep the focus and scope limited, we can do a followup PR if we want to improve readability